extensions.yaml

```diff
@@ -0,0 +1,35 @@
+# RPMs as operating system extensions, distinct from the base ostree commit/image
+# https://github.com/openshift/enhancements/blob/master/enhancements/rhcos/extensions.md
+# and https://github.com/coreos/fedora-coreos-tracker/issues/401
+# We currently explicitly list dependencies for each extension; see
+# https://github.com/coreos/rpm-ostree/issues/2055
+
+extensions:
+  # https://github.com/coreos/fedora-coreos-tracker/issues/326
+  usbguard:
+    - usbguard
+    - libqb
+    - protobuf
+  # kernel-devel
+  # https://github.com/kmods-via-containers/kmods-via-containers/issues/3
+  # https://gitlab.cee.redhat.com/coreos/redhat-coreos/merge_requests/866
+  kernel-devel:
+    - kernel-devel
+    - kernel-core
+    - kernel-headers
+    - kernel-modules
+    - kernel-modules-extra
+
+# Extensions only available on a particular architecture
+extensions-archdep:
```

Comment:
If we're defining a new format anyway, WDYT about folding this into `extensions` instead and make them each richer objects? E.g.

```yaml
extensions:
  usbguard:
    arches: [...]         # optional; defaults to all arches
    exclude_arches: [...] # optional; defaults to none
    packages: [...]       # required
  ...
```

Feels more "YAML-y".

Good as is too though if you want to tweak the format separately (though it's easier to do before it's in production).

Reply:
This would be useful for example when adding more specialized things like matching kernel-headers versions with the one in the host. Something like:

```yaml
kernel-devel:
  packages:
    - kernel-devel
    - kernel-core
    - kernel-headers
    - kernel-modules
    - kernel-modules-extra
  match-evra-of-base-pkg: kernel
```

And then the script would know that when calling `yumdownload` it would append the EVRA of the base package `kernel` to the CLI invocation.

Reply:
Definitely, it's worth getting this format "right" since it becomes an API/ABI of sorts - not impossible to change but still.

I like the first suggestion.  The kernel thing seems way too special case, I'd vote to just hardcode the logic for that one.  If we could think of another case where we need EVRAs matching perhaps it could be declarative.